include/rocksdb/env.h

```diff
@@ -452,9 +452,15 @@ class Env {
   // Sleep/delay the thread for the prescribed number of micro-seconds.
   virtual void SleepForMicroseconds(int micros) = 0;
 
-  // Get the current host name.
+  // Get the current host name as a null terminated string iff the string
+  // length is < len. The hostname should otherwise be truncated to len.
   virtual Status GetHostName(char* name, uint64_t len) = 0;
 
+  // Get the current hostname from the given env as a null terminated
+  // string in result. The result may be truncated if the hostname is too
+  // long
+  static Status GetHostName(Env* env, std::string* result);
```

Comment:
It seems a little weird to me making this static. If making it non-static, virtual or not, you could use the name `GetHostNameString` to avoid compiler issues in combining overloading and overriding. Up to you.